### DISCUSSION TITLE
help with starting values

### DISCUSSION THREAD

**Discussion by WillamysC** [oemof/tespy#438] [Q&A] [answered]
I've been trying the package a while now, have solved simple problems, but my biggest issue is with starting values. Is there any advice or a comum logic for setting them? I have read the "How to Generate Stable Starting Values" page in the documentation, and have used the coolprop library to generate other properties based on the ones found the problems description (mostly pressure/temperature) to fill in as input, but almost always I get it wrong. 

Now, my latest attempt was solving this system:
![image](https://github.com/oemof/tespy/assets/98623054/5edabaae-170e-440a-b2c5-50296a2b0e60)

wich uses R134a, m=0.05kg/s, operates under the pressure limite of 0,8 MPa e 0,14 MPa, and the flash chamber at 0,32MPa. 

the code: 
```python
from tespy.networks import Network
from tespy.components import (
    CycleCloser, Compressor, Condenser, Valve, HeatExchangerSimple, Splitter, Merge
)
from tespy.connections import Connection 
import CoolProp.CoolProp as CP

fluid_list = ['R134a']
my_plant = Network(fluids=fluid_list)

my_plant.set_attr(T_unit='C', p_unit='Pa', h_unit='J / kg', m_unit="kg / s", s_unit='J / kgK')

cc = CycleCloser('cycle closer')
ev = HeatExchangerSimple('evaporador')
cp_bp = Compressor('compressor baixa pressao')
va1 = Valve('valva de expansao 78')
sl = Splitter('separador de liq')

me = Merge("merge")
cp_ap = Compressor('compressor alta presao')
va2 = Valve('valva de expansao 56')
co = HeatExchangerSimple('condensador')

c1 = Connection(cc, 'out1', cp_bp, 'in1', label='1')
c2 = Connection(cp_bp, 'out1', me, 'in1', label='2')
c9 = Connection(me, 'out1', cp_ap, 'in1', label='9')
c4 = Connection(cp_ap, 'out1', co, 'in1', label='4')
c5 = Connection(co, 'out1', va2, 'in1', label='5')

c6 = Connection(va2, 'out1', sl, 'in1', label='6')
c3 = Connection(sl, 'out1', me, 'in2', label='3')
c7 = Connection(sl, 'out2', va1, 'in1', label='7')
c8 = Connection(va1, 'out1', ev, 'in1', label='8')
c0 = Connection(ev, 'out1', cc, 'in1', label='0')

my_plant.add_conns(c1, c2, c3, c4, c0, c5, c6, c7, c8, c9)

my_plant.check_network()

co.set_attr(pr=1)
# ev.set_attr(pr=1)

p1 = 140000 #Pa
h1 = CP.PropsSI('H', 'P', p1, 'Q', 1, fluid_list[0])
s1 = CP.PropsSI('S', 'P', p1, 'H', h1, fluid_list[0])
c1.set_attr(m=0.05, p=p1)#, s=s1)

p2 = 320000 #Pa
s2 = s1
h2 = CP.PropsSI('H', 'P', p2, 'S', s2, fluid_list[0])
c2.set_attr(fluid={'R134a': 1}, h=h2)

p3 = p2
h3 = CP.PropsSI('H', 'P', p3, 'Q', 1, fluid_list[0])
s3 = CP.PropsSI('S', 'P', p3, 'H', h3, fluid_list[0])
# c3.set_attr(h=h3)

p4 = 800000 #Pa
h4 = CP.PropsSI('H', 'P', p4, 'Q', 1, fluid_list[0])
T4 = CP.PropsSI('T', 'P', p4, 'H', h4, fluid_list[0])
c4.set_attr(p=p4,x=1)

p5 = p4 #Pa
h5 = CP.PropsSI('H', 'P', p5, 'Q', 0, fluid_list[0])
s5 = CP.PropsSI('S', 'P', p5, 'H', h5, fluid_list[0])
T5 = CP.PropsSI('T', 'P', p5, 'H', h5, fluid_list[0])
c5.set_attr(p=p5, x=0)

h6 = h5
c6.set_attr(h=h6)

p7 = p2
h7 = CP.PropsSI('H', 'P', p7, 'Q', 0, fluid_list[0])
s7 = CP.PropsSI('S', 'P', p7, 'H', h7, fluid_list[0])
T7 = CP.PropsSI('T', 'P', p7, 'H', h7, fluid_list[0])
c7.set_attr(x=0)

h8 = h7
p8 = p1
T8 = CP.PropsSI('T', 'P', p8, 'H', h8, fluid_list[0])
c8.set_attr(p=p8)

c9.set_attr()

my_plant.solve(mode='design')
```

if anyone can help me understand what am I doing wrong, i'd really appreciate..Thank you!

**Comment by fwitte** [comment]
@WillamysC hi!

So there is a couple of things here:

- Instead of the splitter you need to use a droplet separator. A splitter cannot separate liquid from gas phase, a droplet separator does exactly that.
- Some of your specifications lead to a linear dependency (independent of starting values). That is why I usually recommend building a system step by step to better understand which specifications are working and which ones are not. Specifically,
  - x=0 after condenser and h=h6 after the valve, because the valve already imposes enthalpy equality. Remove h6 specification here.
  - p4 and p5 cannot be set at the same time while the condenser has any specified pressure ratio
  - You are missing some information to determine the intermediate pressure level, i.e. any of p2/3/9.

Best

**Reply by WillamysC** [reply]
Hey! Thank you for the notes! I made the changes, but still not able to make it work. Might be missing something. I looked for the english version of the problema, but couldn't find any, so I could show here. Anyways.. 'bigger' the system, harder it gets i think, right? hahaha. Best regards from Brazil!

**Reply by fwitte** [reply, marked as answer]
Then try it with this one:

```python
[...]

co.set_attr(pr=1)

p1 = 140000 #Pa
h1 = CP.PropsSI('H', 'P', p1, 'Q', 1, fluid_list[0])
s1 = CP.PropsSI('S', 'P', p1, 'H', h1, fluid_list[0])
c1.set_attr(m=0.05, p=p1, h=h1)

p2 = 320000
s2 = s1
h2 = CP.PropsSI('H', 'P', p2, 'S', s2, fluid_list[0])
c2.set_attr(fluid={'R134a': 1}, h=h2)

p3 = p2
h3 = CP.PropsSI('H', 'P', p3, 'Q', 1, fluid_list[0])
s3 = CP.PropsSI('S', 'P', p3, 'H', h3, fluid_list[0])
c3.set_attr(p=p3)

p4 = 800000 #Pa
h4 = CP.PropsSI('H', 'P', p4, 'Q', 1, fluid_list[0])
T4 = CP.PropsSI('T', 'P', p4, 'H', h4, fluid_list[0])
c4.set_attr(p=p4,x=1)

p5 = p4 #Pa
h5 = CP.PropsSI('H', 'P', p5, 'Q', 0, fluid_list[0])
s5 = CP.PropsSI('S', 'P', p5, 'H', h5, fluid_list[0])
T5 = CP.PropsSI('T', 'P', p5, 'H', h5, fluid_list[0])
c5.set_attr(x=0)

h6 = h5
c6.set_attr()

p7 = p2
h7 = CP.PropsSI('H', 'P', p7, 'Q', 0, fluid_list[0])
s7 = CP.PropsSI('S', 'P', p7, 'H', h7, fluid_list[0])
T7 = CP.PropsSI('T', 'P', p7, 'H', h7, fluid_list[0])

h8 = h7
p8 = p1
T8 = CP.PropsSI('T', 'P', p8, 'H', h8, fluid_list[0])
c8.set_attr(p=p8)

c9.set_attr()

my_plant.solve(mode='design')
my_plant.print_results()
```

If that does not work there might be some bug with your installation. it works on my side...

**Reply by WillamysC** [reply]
It worked! Thank you very much! The results are slightly different from the book, but I think it might be something with the connections. Sense c9 connects with c4 through cp_ap, they should have the same entropy, as seen bellow:

![image](https://github.com/oemof/tespy/assets/98623054/bc27e275-6b53-4582-ad75-d1b93ae3b3b5)

![image](https://github.com/oemof/tespy/assets/98623054/624dd71d-64f7-40d7-9051-fea4700a8e94)

I'll try figure this out.

**Reply by fwitte** [reply]
I confused 3 and 7, so for the droplet separator the two connection ids (out1 and out2) have to be switched.

**Reply by WillamysC** [reply]
hey! Sorry to bother you once again with that, I switched the outs of the droplet all fine and, so far so good all connections but c4 have the expected values, as you can see:
![image](https://github.com/oemof/tespy/assets/98623054/6b48509a-7f1c-4c2f-95d0-a7867f79348e)

I understand that the values for c4 diverge from the Diagram T-s because both compressors are considered isentropic there, but the balance equations doesn't see it for c9 and c4.. Is that right? Am I thinking straight here, does it make sense?
